cli/src/main/java/com/devonfw/tools/ide/property/Property.java

```diff
@@ -449,21 +451,24 @@ public boolean matches(String nameOrAlias) {
 
   /**
    * @return {@code true} if this {@link Property} is valid, {@code false} if it is {@link #isRequired() required} but no {@link #getValue() value} has been
-   * set.
-   * @throws RuntimeException if the {@link #getValue() value} is violating given constraints. This is checked by the optional {@link Consumer} function given
-   * at construction time.
+   *     set.
+   * @throws RuntimeException if the {@link #getValue() value} is violating given constraints. This is checked by the optional {@link Consumer} function
+   *     given at construction time.
    */
-  public boolean validate() {
+  public ValidationState validate() {
```

Comment:
From the outside we do not need to do any changes on the state.
To express this we should use the interface that was designed for this purpose.
```suggestion
  public ValidationResult validate() {
```